### ISSUE TITLE
Four eyes principle  when retrieving secrets

### ISSUE THREAD

**Issue by iulianroman** [hashicorp/vault#19214]
**Is your feature request related to a problem? Please describe.**
In a very secure and strict environment  we would like to have more approvals/passwords when requesting a specific secret (after authentication). 

**Describe the solution you'd like**
For retrieving specific data we would like to have a four eyes principle: 2 users should approve the request. A concept similar to unseal the vault but for "unsealing the secrets".
**Describe alternatives you've considered**
Did not try any other alternatives. 

**Explain any additional use-cases**
If there are any use-cases that would help us understand the use/need/value please share them as they can help us decide on acceptance and prioritization.

**Additional context**

**Comment by maxb** [comment]
Vault Enterprise includes the concept of "Control Groups" which are a mechanism for approvals.

However, it is not without caveats.

Who can approve a request, is determined by group membership. However if a user who makes a request is also a member of the approval group, they can approve their own requests, defeating the security.

This can to some extent be mitigated by use of Sentinel policies, but this then puts you in the position of having to configure two parts of the control in different ways, and it is relatively easy to fail to do this correctly, resulting in the control not being enforced.

Furthermore, the Sentinel policy language lacks a way to test whether a control group has been applied to a request, without also triggering response wrapping behaviour as a side effect. This means that if you use a Sentinel EGP to mitigate self approval, it can be tricky to avoid imposing unwanted response wrapping on automated processes which should be allowed to read the same secret without approval.

If you are operating in an environment where one human may be able to log in to Vault as multiple identities (e.g. regular and elevated privileges), you have a further threat of one user being able to place a request with one identity and approve with the other, unless you add more governance to prevent this.

The control groups feature lacks any way for a user to review the open requests pending their approval - instead the requesting user must communicate a request ID out of band to the approving user.

The approving user is shown some information about the request they are about to approve, but not as much as I would like.

Once a four eyes process has been approved, you still have a situation where the approved retriever now has possession of the secret, and you need to build your own policies and procedures so they can't write it down and re-use it forever. (Ideally, it should be single-use.)

----

In short: four eyes is really hard to get right, and whilst Vault (Enterprise) has some relevant parts, there is still a lot of effort left to the individual implementor, and even then there are still rough edges.
